docs/advanced-features/event-driven-packaging.md

```diff
@@ -0,0 +1,298 @@
+            "ProjectID": "YYD",
+            "StudyID": "ABC-23-023394",
+            "Target": "T1",
+            "Test Material": "FIS-1111",
+            "Tissue": "Lungs"
+        }
+
+        with tempfile.TemporaryDirectory() as tmpdir:
+            tmpdir_path = pathlib.Path(tmpdir)
+            for name, body in beautify_files.items():
+                if name in pkg:
+                    logger.debug(f"File {name} already exists. Ignoring.")
+                    continue
+                logger.debug(f"File {name} does not exist at {prefix}. Creating.")
+                file_path = tmpdir_path / name
+                file_path.write_text(body)
+                pkg.set(name, file_path)
+        
+            # Add metadata to package
+            pkg.set_meta(meta)
+            # Remove leading & trailing characters
+            pkg_name = prefix.strip("/")
+        
+            # Define callable Quilt push()
+            push = functools.partial(
+                pkg.push,
+                pkg_name,
+                registry=f"s3://{bucket}",
+                force=True,
+                message="Created by EDP",
+                workflow=WORKFLOW_NAME
+            )
+        
+            # Validate against the Quilt workflow schema
+            try:
+                push(dedupe=True)
+            except quilt3.workflows.WorkflowValidationError as e:
+                logger.warning("Workflow check failed")
+            
+                # Write out error to README.md file in quarantine bucket
+                file_path = tmpdir_path / "README.md"
+                file_path.write_text(str(e))
+                pkg.set("README.md", file_path)
+            
+                # Push package to quarantine bucket
+                push(registry=f"s3://{QUARANTINE_BUCKET_NAME}", workflow=...)
+            
+                # Error SNS notification content
+                subject = f"Failed validation for package {pkg_name}"
+                if len(subject) > MAX_SNS_SUBJECT_LEN:
+                    subject = subject[: MAX_SNS_SUBJECT_LEN - 1] + "…"
+                message = (
+                    f"Validation failed for workflow {WORKFLOW_NAME} while pushing "
+                    f"package with name {pkg_name} to {bucket}. It was pushed to "
+                    f"{QUARANTINE_BUCKET_NAME} instead.\n"
+                    f"{QUILT_URL}/b/{QUARANTINE_BUCKET_NAME}/packages/{pkg_name}\n\n"
+                    f"Error message is:\n{e}\n"
+                )
+                # Publish notification to SNS topic
+                sns.publish(
+                    TopicArn=SNS_TOPIC_ARN,
+                    Message=message,
+                    Subject=subject,
+                )
+    ```
+1. If a metadata validation error occurs (and they are subscribed
+to the `SNS_TOPIC_ARN`) the Lab and Computational scientists receive
+an email notification of the error and can check the package created in the
+quarantine bucket for problems.
```

Comment:
This is a bit unclear. I think the point is that SNS can email users. Primarily "subscribed" is confusing and does not necessarily imply email. So I would focus on the SNS events and then add a note that these can be trafficked over email.

Reply:
This is an actual use case and not unclear in the context of the example and the steps above:

```
1. A custom SNS topic is created for monitoring data package creation
that Lab and Computational scientists subscribe to (`SNS_TOPIC_ARN`).
```

I've made it more clear.